andela/bp-esa-backend
# Minifies the served application when  using docker

#### What does this PR do?
Ensures that only the necessary content is copied when creating an image.

#### Description of Task to be completed?
- Remove unnecessary directories from the docker images
- Update how the application is started
- Specify image to use  when using  `docker-compose`

#### How should this be manually tested?
 Ensure that you have [docker installed](https://www.docker.com/)
 On the terminal run the following  commands:
1. Change to the application root directory
2. Uncomment  [this line](https://github.com/andela/bp-esa-backend/blob/1828ebf586cf5a04cae00b5b92fda845f3825932/Dockerfile#L23) for  local development. Change  the  following Environment Variables:
>REDIS_HOST=redis

3. Build a docker image with the following command
` docker build -t esa-backend  .`
4. Use docker  compose to install services required by the application i.e *postgres* and *redis*
5. Then run `docker-compose up`

#### Any background context you want to provide?
Before the change, the served application on container registry was large ( *1.1GB*). 
This changes  reduce the image sixe  to *180MB* . 
This means that will use considerably less resources to serve contaiiners created from this image.

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/162951587

#### Postman Documentation


#### Screenshots (If applicable)
